certdepot/cert_request.go

```diff
@@ -0,0 +1,119 @@
+package certdepot
+
+import (
+	"regexp"
+	"strings"
+
+	"github.com/mongodb/grip"
+	"github.com/pkg/errors"
+	"github.com/square/certstrap/depot"
+	"github.com/square/certstrap/pkix"
+)
+
+type CertRequestOptions struct {
+	// The depot which stores the id/cert.
+	Depot depot.Depot
+	// Passprhase to encrypt private-key PEM block.
+	Passphrase string
+	// Size (in bits) of RSA keypair to generate (defaults to 2048).
+	KeyBits int
+	// Sets the Organization (O) field of the certificate.
+	Organization string
+	// Sets the Country (C) field of the certificate.
+	Country string
+	// Sets the Locality (L) field of the certificate.
+	Locality string
+	// Sets the Common Name (CN) field of the certificate.
+	CommonName string
+	// Sets the Organizational Unit (OU) field of the certificate.
+	OrganizationalUnit string
+	// Sets the State/Province (ST) field of the certificate.
+	Province string
+	// IP addresses to add as subject alt name.
+	IP []string
+	// DNS entries to add as subject alt name.
+	Domain []string
+	// URI values to add as subject alt name.
+	URI []string
+	// Path to private key PEM file (if blank, will generate new keypair).
+	Key string
+}
+
+// NewCertRequest creates a new certificate (CSR).
+func NewCertRequest(opts CertRequestOptions) error {
```

Comment:
split this function up to make it more testable? 

also does it make sense to take the Depot out of the Options struct and then have a method that's like Generate(Deopot) error to do this function?